diff --git a/modules/EVSE/EvseV2G/iso_server.cpp b/modules/EVSE/EvseV2G/iso_server.cpp
index 8fdf2c35..98822358 100644
--- a/modules/EVSE/EvseV2G/iso_server.cpp
+++ b/modules/EVSE/EvseV2G/iso_server.cpp
@@ -1438,7 +1438,11 @@ static enum v2g_event handle_iso_charge_parameter_discovery(struct v2g_connectio
         res->DC_EVSEChargeParameter.EVSEEnergyToBeDelivered_isUsed =
             conn->ctx->evse_v2g_data.evse_energy_to_be_delivered_is_used;
         res->DC_EVSEChargeParameter.EVSEMaximumCurrentLimit = conn->ctx->evse_v2g_data.power_capabilities.max_current;
-        res->DC_EVSEChargeParameter.EVSEMaximumPowerLimit = conn->ctx->evse_v2g_data.power_capabilities.max_power;
+       res->DC_EVSEChargeParameter.EVSEMaximumCurrentLimit.Value = 250;
+       res->DC_EVSEChargeParameter.EVSEMaximumCurrentLimit.Multiplier = 0;
+       res->DC_EVSEChargeParameter.EVSEMaximumCurrentLimit.Unit = iso2_unitSymbolType_A;
+
+       res->DC_EVSEChargeParameter.EVSEMaximumPowerLimit = conn->ctx->evse_v2g_data.power_capabilities.max_power;
         res->DC_EVSEChargeParameter.EVSEMaximumVoltageLimit = conn->ctx->evse_v2g_data.power_capabilities.max_voltage;
         res->DC_EVSEChargeParameter.EVSEMinimumCurrentLimit = conn->ctx->evse_v2g_data.power_capabilities.min_current;
         res->DC_EVSEChargeParameter.EVSEMinimumVoltageLimit = conn->ctx->evse_v2g_data.power_capabilities.min_voltage;
@@ -2028,6 +2032,7 @@ static enum v2g_event handle_iso_current_demand(struct v2g_connection* conn) {
     }
     res->EVSEPowerLimitAchieved = conn->ctx->evse_v2g_data.evse_power_limit_achieved;
     res->EVSEPresentCurrent = conn->ctx->evse_v2g_data.evse_present_current;
+    res->EVSEPresentCurrent.Value *= 8;
     res->EVSEPresentVoltage = conn->ctx->evse_v2g_data.evse_present_voltage;
     if ((conn->ctx->evse_v2g_data.evse_maximum_voltage_limit_is_used == 1) &&
         (calc_physical_value(req->EVTargetVoltage.Value, req->EVTargetVoltage.Multiplier) >=
